### ISSUE TITLE
Error in "Ensure that {{ openshift_master_ensure_nodes_service }} has nodes to run on"

### ISSUE THREAD

**Issue by AllanCano** [openshift/openshift-ansible#7697]
#### Description

In running the deploy the playbook died in the step "Ensure that {{ openshift_master_ensure_nodes_service }} has nodes to run on"

The traceback shows the problem is in the oc call from:

/etc/ansible/playbooks/openshift-ansible/roles/openshift_master/tasks/ensure_nodes_matching_selector.yml

The argument --field-selector is not valid.  Change this in the file from "field_select" to simply "selector", and yes I'm assuming these are interchangeable arguments, and things progress.

##### Version

- oc verions
```
# oc version
oc v3.7.1+c2ce2c0-1
kubernetes v1.7.6+a08f5eeb62
features: Basic-Auth GSSAPI Kerberos SPNEGO
```
- Ansible verions 
```
# ansible --version
ansible 2.6.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```
- Ansible Openshift Playbooks v3.9

- CentOS version
```
# cat /etc/redhat-release
CentOS Linux release 7.4.1708 (Core)
```
- Openshift version 
```
# openshift version
openshift v3.7.1+c2ce2c0-1
kubernetes v1.7.6+a08f5eeb62
etcd 3.2.8
```

##### Inventory Hosts Files
```
#---- Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_ssh_user=root
debug_level='2'
containerized='false'
openshift_deployment_type=origin
os_firewall_use_firewalld='true'

#-- Configure networking
#openshift_master_default_subdomain=fossil.com
openshift_use_openshift_sdn=true
#openshift_use_flannel=true
#flannel_interface=bond0

# Uncomment the following to enable htpasswd authentication; defaults to DenyAllPasswordIdentityProvider.
#openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]

# Native high availbility cluster method with optional load balancer.
# If no lb group is defined installer assumes that a load balancer has
# been preconfigured. For installation the value of
# openshift_master_cluster_hostname must resolve to the load balancer
# or to one or all of the masters defined in the inventory if no load
# balancer is present.
#-- When we get to production use an external LB
openshift_master_cluster_method=native
#openshift_master_cluster_hostname=uusrcdkub00.fossil.com
#openshift_master_cluster_public_hostname=uusrcdkub00.fossil.com

# apply updated node defaults
openshift_node_kubelet_args={'pods-per-core': ['10'], 'max-pods': ['250'], 'image-gc-high-threshold': ['90'], 'image-gc-low-threshold': ['80']}

# enable ntp on masters to ensure proper failover
openshift_clock_enabled=true

# Setup rolling updates at the service level
openshift_rolling_restart_mode='services'

# Change all the certificates to last for 10 years.
openshift_hosted_registry_cert_expire_days='720'
openshift_ca_cert_expire_days='720'
openshift_node_cert_expire_days='720'
openshift_master_cert_expire_days='720'
etcd_ca_default_days='720'


#---- host group for masters
[masters]
uusrcdkub01.fossil.com
uusrcdkub02.fossil.com
uusrcdkub03.fossil.com

#---- host group for etcd
[etcd]
uusrcdkub01.fossil.com
uusrcdkub02.fossil.com
uusrcdkub03.fossil.com

#---- Specify load balancer host
#[lb]
#uusrcdkub00.fossil.com

#---- host group for nodes, includes region info
[nodes]
uusrcdkub0[1:3].fossil.com openshift_schedulable=true       openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
uusrcdkub0[4:6].fossil.com openshift_schedulable=true       openshift_node_labels="{'region': 'NAUS', 'zone': 'CS'}"

# Create an OSEv3 group that contains the master, nodes, etcd, and lb groups.
# The lb group lets Ansible configure HAProxy as the load balancing solution.
# Comment lb out if your load balancer is pre-configured.
[OSEv3:children]
masters
nodes
etcd
#lb
```
##### Steps To Reproduce

- After base CentOS 7.3 install

1. Install pre-reqs and update the OS 
```
yum -y install wget git net-tools bind-utils iptables-services bridge-utils bash-completion kexec-tools sos psacct curl docker etcd atomic pyOpenSSL.x86_64 python-cryptography python-lxml
yum update
systemctl reboot
```
2. Increase the root volume group to use the complete disk 
```
fdisk /dev/sda
d
2
n
primary
2
default
default

systemctl reboot
pvresize /dev/sda2
```
3. Update the docker config to use a LV for the docker storage 
```
systemctl stop docker
rm -rf /var/lib/docker
vi /etc/sysconfig/docker-storage-setup
# File contents below
/bin/docker-storage-setup
systemctl start docker
systemctl enbale docker
docker info
lvs
```

File: /etc/sysconfig/docker-storage-setup
```
# Edit this file to override any configuration options specified in
# /usr/lib/docker-storage-setup/docker-storage-setup.
#
# For more details refer to "man docker-storage-setup"
#STORAGE_DRIVER
#DEVS=/dev/sda
VG=centos
GROWPART=enable
AUTO_EXTEND_POOL=enable
MIN_DATA_SIZE=8G
POOL_AUTOEXTEND_THRESHOLD=60
POOL_AUTOEXTEND_PERCENT=10
```
	
- Install Ansible on the first master
We need at least Ansible 2.4.3.  At the time of writing this doc the latest published was 2.4.2.  So, we need to role our own RPM. To build Ansible we need:
```
	yum -y install asciidoc rpm-build python2-devel
```
- Build and install the latest Ansible 
```
	mkdir Development
	git clone https://github.com/ansible/ansible.git
	cd ansible
	make rpm
	cd /root/Development/ansible/rpm-build
	rpm --install ansible-2.6.0-100.git201803142047.0cf2ecb.devel.el7.centos.noarch.rpm
```
- Pull the OpenShift Ansible Source 
```
	cd /etc/ansible/
	mkdir playbooks
	cd playbooks
	git clone https://github.com/openshift/openshift-ansible
```
- Create the Ansible Host File 
```
  vi /etc/ansbile/hosts
```
File contents listed above.

- Run the pre-reqs and deploy
```
  cd /etc/ansible/playbooks/openshift-ansible/playbooks/
  ansible-playbook -i /etc/ansible/hosts $PWD/prerequisites.yml
	
  cd /etc/ansible/playbooks/openshift-ansible/playbooks/openshift-checks
  vi roles/openshift_health_checker/openshift_checks/package_version.py
```  
I updated the allowed docker versions for (3,7) to include 1.13 and 1.13.1
```
  ansible-playbook -i /etc/ansible/hosts $PWD/pre-install.yml
    
  cd /etc/ansible/playbooks/openshift-ansible/playbooks/
  ansible-playbook -vvv -i /etc/ansible/hosts $PWD/deploy_cluster.yml
```

##### Expected Results
Describe what you expected to happen.

```
I expected not to have to alter the file to get the oc command to run
```

##### Observed Results
The deploy_cluster playbook terminate at "Ensure that {{ openshift_master_ensure_nodes_service }} has nodes to run on" and noted in the verbose output the previous task terminated with the results of a failed oc.

I forgot to cut and paste the exact error and after fixing the parameter "field_select" to "select" the playbook has finished without issue.

Don't know how to raise a bug report so hopefully this gets to someone who does.

Cheers.

**Comment by RafnexJr** [comment]
I had the same issue, with the same resolution.

**Comment by vrutkovs** [comment]
>Ansible Openshift Playbooks v3.9
>oc v3.7.1+c2ce2c0-1

Don't use playbooks for 3.9 version to setup 3.7. These playbooks use several new `oc` CLI options, which are available in 3.9 only - e.g. `field_select`. In order to install 3.7 you should use `release-3.7` branch
